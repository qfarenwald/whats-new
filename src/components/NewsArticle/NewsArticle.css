.news-container {
  width: 50%;
}

img {
  padding: 20px 0px;
  width: 100%;
}

p {
  padding: 20px 0px;
}
