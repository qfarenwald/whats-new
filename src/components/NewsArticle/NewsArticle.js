import React from 'react';
import './NewsArticle.css';

const NewsArticle = ( {img, headline, description, url} ) => {
  return (
    <div className="article">
      <div className="img-container">
        <img src={img} alt="" />
      </div>
      <article>
        <h3>{headline}</h3>
        <p>{description}</p>
        <a href={url}>Full Story</a>
      </article>
    </div>
  )
}

export default NewsArticle;
