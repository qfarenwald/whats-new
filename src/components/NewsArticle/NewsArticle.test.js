import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  it('should match the snapshot with all data pass in correctly', () => {
    const wrapper = shallow(
      <NewsArticle key={1}
        img='https://theknow.denverpost.com/wp-content/uploads/2016/03/CDXXTHEWHODSC_4082x.jpg'
        headline='The Who postpones Denver Concert at the Pepsi Center'
        description='Classic-rock band The Who has postponed its Denver concert, set for Sunday, Sept. 29, after an illness robbed singer Roger Daltrey of his voice, the band said on Twitter.'
        url='https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/'
      />
    )
    expect(wrapper).toMatchSnapshot();
  })

  // it('should link to another page when link is clicked', () => {
  //   const wrapper = shallow(
  //     <NewsArticle key={1}
  //       img='https://theknow.denverpost.com/wp-content/uploads/2016/03/CDXXTHEWHODSC_4082x.jpg'
  //       headline='The Who postpones Denver Concert at the Pepsi Center'
  //       description='Classic-rock band The Who has postponed its Denver concert, set for Sunday, Sept. 29, after an illness robbed singer Roger Daltrey of his voice, the band said on Twitter.'
  //       url='https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/'
  //     />
  //   )
  //
  //   wrapper.find('a').simulate('click')
  //
  //   expect(wrapper).toHaveBeenCalledWith('https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/')
  // })

})
