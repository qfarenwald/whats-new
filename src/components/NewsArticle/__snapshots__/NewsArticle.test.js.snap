// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all data pass in correctly 1`] = `
<div
  className="article"
>
  <div
    className="img-container"
  >
    <img
      alt=""
      src="https://theknow.denverpost.com/wp-content/uploads/2016/03/CDXXTHEWHODSC_4082x.jpg"
    />
  </div>
  <article>
    <h3>
      The Who postpones Denver Concert at the Pepsi Center
    </h3>
    <p>
      Classic-rock band The Who has postponed its Denver concert, set for Sunday, Sept. 29, after an illness robbed singer Roger Daltrey of his voice, the band said on Twitter.
    </p>
    <a
      href="https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/"
    >
      Full Story
    </a>
  </article>
</div>
`;
